lxd/storage/drivers/driver_zfs_volumes.go

```diff
@@ -1657,10 +1705,15 @@ func (d *zfs) MountVolume(vol Volume, op *operations.Operation) error {
 	defer revert.Fail()
 
 	dataset := d.dataset(vol, false)
+	mountPath := vol.MountPath()
+
+	volumeType, err := d.getDatasetProperty(dataset, "type")
```

Comment:
We should be able to use the volume DB config to decide this and avoid running the extra command (which will have an impact on performance at scale).

Reply:
I have been looking into this but cannot seem to get the tests to succeed when using `vol.config`.

Reply:
What issues are you still having with this? It would be good to avoid the need to call `getDatasetProperty` and just use the volume's config.